### PR TITLE
set up documentation using Documenter.jl

### PR DESCRIPTION
**Pull request by fzierler**
This sets up the file structure for automatically creating docs using Documenter.jl. As an example I have added docs for the autocorrelation functions. To create docs locally make sure that Documenter.jl is added to your global environment (the same environment as e.g. Revise) and run the docs/make.jl file. The documentation can then be found at docs/build/index.html

### CODE REVIEW

File: src/statistics.jl [hunk 1]
```diff
@@ -1,28 +1,46 @@
-# Autocorrelation for a specifig lag
-function autocor(x::AbstractVector, lag::Integer)
+"""
+    autocor(x,lag)
+
+Returns autocorrelation function for a distance of lag, i.e. between values of
+```x[t]``` and ```x[t+lag]```. The autocorrelation function is normalized such
+that ```autocor(x,0) = 1```. This is equvivalent to the quantity
+``\\Gamma_X(t)`` of equation (4.61) in Gattringer/Lang.
```
Comment (briederer) [inline src/statistics.jl]: Better citation of the book would be nice.
Is there anything in documenter.jl to have something like a "Reference"-section with Bibliography to link to?

File: src/statistics.jl [hunk 2]
```diff
@@ -1,28 +1,46 @@
-# Autocorrelation for a specifig lag
-function autocor(x::AbstractVector, lag::Integer)
+"""
+    autocor(x,lag)
```
Comment (briederer) [inline src/statistics.jl:2]: Any specific reason I miss for calling it `lag`?

Comment (fzierler) [reply src/statistics.jl:2]: It is the same name that is being used in StatsBase.jl - we could call it something else. Calling it ```time``` or ```t``` could be misleading but maybe something like ```steps``` or ```timesteps```?

Comment (briederer) [reply src/statistics.jl:2]: Okay yes makes sense to stick to something that has already been around for some time.

File: src/statistics.jl (hunk 1 above)
Comment (fzierler) [reply src/statistics.jl]: It turns out that there is a dedicated plug-in package for this [DocumenterCitations.jl](https://github.com/ali-ramadhan/DocumenterCitations.jl) . I had to modify the standard .bib entries generated by inspire for it to work and the bibliography style cannot be customized (yet) (see https://github.com/ali-ramadhan/DocumenterCitations.jl/issues/22 ) but this should do it for now.

**Comment by briederer** [comment]
I am fine with merging for now.
Thoug I still would suggest to use a separate `Project.toml` for the docs folder to keep src clear (see for instance [PlotDocs.jl](https://github.com/JuliaPlots/PlotDocs.jl))

**Comment by fzierler** [comment]
I have added a separate ```Project.toml``` file. I don't know if CompatHelper will work for this? I am also happy with merging this for now.

**Comment by briederer** [comment]
Good question.
Seems like all the julia projects I saw which use these kind of separation don't have any `[compat]` sections at all in the `docs/Project.toml`.
Will open an issue for further discussion and merge the branch.

**Pull request merged**